src/Microsoft.DotNet.CodeFormatting/FormattingEngineImplementation.cs

```diff
@@ -316,7 +357,31 @@ private Solution RemoveTablePreprocessorSymbol(Solution newSolution, Solution ol
 
             var compilation = await project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
 
-            IEnumerable<DiagnosticAnalyzer> analyzersToRun = _analyzers.Where(a => a.SupportsLanguage(project.Language));
+            // If there are any compilation errors then some of the semantic rules will be faulty. Log the
+            // compiler errors and the metadatareferences to diagnose the issues.
+            if (Verbose)
+            {
+                var compilerDiagnostics = compilation.GetDiagnostics().Where(d => d.Severity != DiagnosticSeverity.Hidden && d.Severity != DiagnosticSeverity.Info);
+                if (compilerDiagnostics.Any())
+                {
+
+                        Console.WriteLine("Error count " + compilerDiagnostics.Count());
+
+                        Console.WriteLine("Metadata References for {0}", project.Name);
+                        foreach (var mr in project.MetadataReferences)
+                        {
+                            Console.WriteLine(mr.Display);
+                        }
+
```

Comment:
Maybe want some separator text or white space between the metadata and the diagnostics?